std/datetime/systime.d

```diff
@@ -11729,7 +11729,7 @@ private @safe:
 
         immutable ot = otherTZ.utcToTZ(0);
 
-        auto diffs = [0L, lt, ot];
+        auto diffs = [0L, cast()lt, cast()ot];
```

Comment:
Don't cast these. If it's a problem that they're `immutable`, then just change their declarations to `auto`. However, I don't see why the change should be necessary.

Reply:
These are  value types, the variables `lt` and `ot` are never changed still even with the cast. The purpose of the cast is to make the array `diffs` mutable such that it can be sorted in-place.

Reply:
Given that the main purpose of these variables is to use them in this array, it's just silly to explicitly cast away `immutable` rather than use `auto` if the compiler isn't going to decide to implicitly convert to remove `immutable`.

Reply:
Still making variables that are not going to change `const` or `immutable` is a good practice.

Reply:
It's dumb if the code is just going to turn around and cast to make them mutable. And any time that `cast()` is used, it's a code smell. There are cases where it makes sense, and it's not actually causing problems here, because the type is a value type, but it's still a code smell any time that it's in code and reason to examine the code more carefully. As such, it's far better to avoid needing such casts.

Reply:
There is a caveat either way you go. `auto` means the variable is likely to change which is not true in this situation, while a `cast()` means something is not right which is correct in this case, the type deduction doesn't result in the needed type hence the `cast()`.

Reply:
It's not like this code needs to protect against changing the variables in question, and `cast()` should almost always be avoided if possible. The code only uses `immutable` here, because I default to `immutable` for variables of primitive types if it doesn't matter. Once it does, I use `auto`, and I'm very much against using `cast()`.